File: 2294-partition-array-such-that-maximum-difference-is-k/2294-partition-array-such-that-maximum-difference-is-k.cs
public class Solution {
    public int PartitionArray(int[] nums, int k) {
        Array.Sort(nums);
        var count = 1;  // 1 = subsequence with all numbers

        var smallest = nums[0];
        foreach(var num in nums){
            if(num - smallest <= k) continue;
            smallest = num;
            count++;    // start new subsequence
        }

        return count;
    }
}

File: 2294-partition-array-such-that-maximum-difference-is-k/2294-partition-array-such-that-maximum-difference-is-k.py
class Solution:
    def partitionArray(self, nums: List[int], k: int) -> int:
        nums.sort()
        count = 1  # 1 = subsequence with all numbers

        smallest = nums[0]
        for num in nums:
            if num - smallest <= k: continue
            smallest = num
            count += 1    # start new subsequence

        return count